src/snowflake/snowpark/session.py

```diff
@@ -1320,6 +1315,27 @@ def _resolve_packages(
     ) -> List[str]:
         # Extract package names, whether they are local, and their associated Requirement objects
         package_dict = self._parse_packages(packages)
+        if isinstance(self._conn, MockServerConnection):
+            # in local testing we don't resolve the packages, we just return what is added
+            errors = []
+            for pkg_name, _, pkg_req in package_dict.values():
+                if (
+                    pkg_name in self._packages
+                    and str(pkg_req) != self._packages[pkg_name]
+                ):
+                    errors.append(
+                        ValueError(
+                            f"Cannot add package '{str(pkg_req)}' because {self._packages[pkg_name]} "
+                            "is already added."
+                        )
+                    )
+                else:
+                    self._packages[pkg_name] = str(pkg_req)
+            if len(errors) == 1:
+                raise errors[0]
+            elif len(errors) > 0:
```

Comment:
Question: why do we need this case?

Reply:
this is copied and pasted from the existing package resolving case: https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/session.py#L1270-L1274

```python
                    except Exception as ex:  # pragma: no cover
                        logging.warning(
                            "Failed to get the local distribution of package %s: %s",
                            package_name,
                            ex,
                        )

            if package_name in current_packages:
                if current_packages[package_name] != package:
                    errors.append(
                        ValueError(
                            f"Cannot add package '{package}' because {current_packages[package_name]} "
                            "is already added."
                        )
                    )
            else:
                current_packages[package_name] = package

        # Raise all exceptions at once so users know all issues in a single invocation.
        if len(errors) == 1:
            raise errors[0]
        elif len(errors) > 0:
            raise RuntimeError(errors)
```